wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wdl/WdlWriterImpl.scala

```diff
@@ -0,0 +1,438 @@
+        case Some(o) => o.toWdlV1
+        case None => ""
+      }
+
+      s"""workflow ${a.name} {
+         |${indent(inputs)}
+         |${indentAndCombine(a.graphElements.map(_.toWdlV1))}
+         |${indent(outputs)}
+         |}""".stripMargin
+    }
+  }
+
+  implicit val runtimeAttributesSectionElementWriter: WdlWriter[RuntimeAttributesSectionElement] = new WdlWriter[RuntimeAttributesSectionElement] {
+    override def toWdlV1(a: RuntimeAttributesSectionElement): String = {
+      val runtimeMap = a.runtimeAttributes map { pair =>
+        s"${pair.key}: ${pair.value.toWdlV1}"
+      }
+
+      s"""runtime {
+         |${indentAndCombine(runtimeMap)}}""".stripMargin
+    }
+  }
+
+  implicit val metaValueElementWriter: WdlWriter[MetaValueElement] = new WdlWriter[MetaValueElement] {
+    override def toWdlV1(a: MetaValueElement): String = a match {
+      case _: MetaValueElementNull.type => "null"
+      case a: MetaValueElementBoolean => a.value.toString
+      case a: MetaValueElementFloat => a.value.toString
+      case a: MetaValueElementInteger => a.value.toString
+      case a: MetaValueElementString => "\"" + a.value + "\""
+      case a: MetaValueElementObject =>
+        "{" + a.value.map { pair =>
+          s"${pair._1}: ${metaValueElementWriter.toWdlV1(pair._2)}"
+        }.mkString(", ") + "}"
+      case a: MetaValueElementArray => "[" + a.value.map(metaValueElementWriter.toWdlV1).mkString(", ") + "]"
+    }
+  }
+
+
+  implicit val metaSectionElementWriter: WdlWriter[MetaSectionElement] = new WdlWriter[MetaSectionElement] {
+    override def toWdlV1(a: MetaSectionElement): String = {
+      val map = a.meta.map { pair =>
+        s"${pair._1}: ${pair._2.toWdlV1}"
+      }
+      s"""meta {
+         |${indentAndCombine(map)}
+         |}""".stripMargin
+    }
+  }
+
+  implicit val parameterMetaSectionElementWriter: WdlWriter[ParameterMetaSectionElement] = new WdlWriter[ParameterMetaSectionElement] {
+    override def toWdlV1(a: ParameterMetaSectionElement): String = {
+      val map = a.metaAttributes.map { pair =>
+        s"${pair._1}: ${pair._2.toWdlV1}"
+      }
+      s"""parameter_meta {
+         |${indentAndCombine(map)}
+         |}""".stripMargin
+    }
+  }
+
+  implicit val taskDefinitionTypeElementWriter: WdlWriter[TaskDefinitionElement] = new WdlWriter[TaskDefinitionElement] {
+    override def toWdlV1(a: TaskDefinitionElement) = {
+      val inputs = a.inputsSection match {
+        case Some(i) => i.toWdlV1
+        case None => ""
+      }
+      val outputs = a.outputsSection match {
+        case Some(o) => o.toWdlV1
+        case None => ""
+      }
+      val runtime = a.runtimeSection match {
+        case Some(r) => r.toWdlV1
+        case None => ""
+      }
+      val meta = a.metaSection match {
+        case Some(m) => m.toWdlV1
+        case None => ""
+      }
+      val parameterMeta = a.parameterMetaSection match {
+        case Some(p) => p.toWdlV1
+        case None => ""
+      }
```

Comment:
don't know how you feel about folding Options but it would be a lot less lines

Reply:
My objection to folding `Option` is that it doesn't actually work in analogy to folding lists, so I just find I have to relearn what it's doing each time. I don't know why they had to override `fold` as a method name rather than calling it `mapOrDefault` or some such. OTOH it makes it terser so up to you...